couchbase/operations/document_query.cxx

```diff
@@ -212,10 +212,10 @@ query_request::make_response(error_context::query&& ctx, const encoded_response_
             meta_metrics.result_size = m->at("resultSize").get_unsigned();
             meta_metrics.elapsed_time = utils::parse_duration(m->at("elapsedTime").get_string());
             meta_metrics.execution_time = utils::parse_duration(m->at("executionTime").get_string());
-            meta_metrics.sort_count = m->template optional<std::uint64_t>("sortCount");
-            meta_metrics.mutation_count = m->template optional<std::uint64_t>("mutationCount");
-            meta_metrics.error_count = m->template optional<std::uint64_t>("errorCount");
-            meta_metrics.warning_count = m->template optional<std::uint64_t>("warningCount");
+            meta_metrics.sort_count = m->template optional<std::uint64_t>("sortCount").value_or(0);
```

Comment:
Is there somewhere that defines which of these are optional? It's not in the RFC

Reply:
Right, for analytics these are not optional, but query still writes them optionally (https://github.com/couchbase/query/blob/f0109b241dd084d8e52ad4f53bcdba8a339492c0/server/http/service_response.go#L510-L520)